lib/events.js

```diff
@@ -224,11 +234,11 @@ function addCatch(that, promise, type, args) {
     const then = promise.then;
 
     if (typeof then === 'function') {
-      then.call(promise, undefined, function(err) {
+      ReflectApply(then, promise, [undefined, function(err) {
         // The callback is called with nextTick to avoid a follow-up
         // rejection from this promise.
         process.nextTick(emitUnhandledRejectionOrErr, that, err, type, args);
-      });
+      }]);
```

Comment:
Isn't the perf regression because of this? There was no array created before. Again, I don't understand the reasoning behind changing instances of `.call` to `ReflectApply` instead of `FunctionPrototypeCall`.

Reply:
I don't think so, this code is never reached by the benchmark scripts that show perf regression.
I worked on this before https://github.com/nodejs/node/pull/36221, at that time `ReflectApply` had less overhead than `FunctionPrototypeCall`. It's not true anymore, but I didn't bother changing it. Do you think it would be clearer with `FunctionPrototypeCall`?

Another path I have thought of would be to check `require('internal/util/types').isPromise(promise)` to call `PromisePrototypeCatch` and fallback to `.then` for non-`Promise` _thenables_.

Reply:
Are you sure this is not reached? The benchmark script calls `on`. So, that's where this backtrace starts:
```js
// line 470
EventEmitter.prototype.on = EventEmitter.prototype.addListener;

// line 466
EventEmitter.prototype.addListener = function addListener(type, listener) {
  return _addListener(this, type, listener, false);
};

function _addListener(target, type, listener, prepend) {
...
// line 419
      target.emit('newListener', type,
                  listener.listener ? listener.listener : listener);
...
}

EventEmitter.prototype.emit = function emit(type, ...args) {
...
// line 382
      addCatch(this, result, type, args);
...
// line 396
        addCatch(this, result, type, args);
}
```

Eventually, there is a call to `addCatch`, right?

Reply:
It's worth trying for sure! Thanks.